simpleclient/src/main/java/io/prometheus/client/CollectorRegistry.java

```diff
@@ -168,10 +168,12 @@ private Iterator<Collector> filteredCollectorIterator() {
       } else {
         HashSet<Collector> collectors = new HashSet<Collector>();
         synchronized (namesCollectorsLock) {
-          for (Map.Entry<String, Collector> entry : namesToCollectors.entrySet()) {
-            // Note that namesToCollectors contains keys for all combinations of suffixes (_total, _info, etc.).
-            if (sampleNameFilter.test(entry.getKey())) {
-              collectors.add(entry.getValue());
+          for (Map.Entry<Collector, List<String>> entry : collectorsToNames.entrySet()) {
+            for (String name : entry.getValue()) {
```

Comment:
Maybe I am missing something but it doesn't seem to fix the issue.
The problem is the `List<String>` of names is empty, so this inner loop is not going to run.
One way to make it work, though I don't know if this is the behavior you want, is something like this:
```java
          for (Map.Entry<Collector, List<String>> entry : collectorsToNames.entrySet()) {
            if (entry.getValue().isEmpty()) {
              collectors.add(entry.getKey());
            } else {
              for (String name : entry.getValue()) {
                if (sampleNameFilter.test(name)) {
                  collectors.add(entry.getKey());
                  break;
                }
              }
            }
          }
```